internal/impl/gcp/output_pubsub.go

```diff
@@ -3,34 +3,22 @@ package gcp
 import (
 	"context"
 	"fmt"
-	"strings"
 	"sync"
-	"time"
-
-	"google.golang.org/api/option"
 
 	"cloud.google.com/go/pubsub"
-
-	"github.com/benthosdev/benthos/v4/internal/batch"
-	"github.com/benthosdev/benthos/v4/internal/bloblang/field"
-	"github.com/benthosdev/benthos/v4/internal/bundle"
-	"github.com/benthosdev/benthos/v4/internal/component"
-	"github.com/benthosdev/benthos/v4/internal/component/metrics"
-	"github.com/benthosdev/benthos/v4/internal/component/output"
-	"github.com/benthosdev/benthos/v4/internal/component/output/processors"
-	"github.com/benthosdev/benthos/v4/internal/docs"
-	"github.com/benthosdev/benthos/v4/internal/log"
-	"github.com/benthosdev/benthos/v4/internal/message"
-	"github.com/benthosdev/benthos/v4/internal/metadata"
+	"github.com/benthosdev/benthos/v4/public/service"
+	"github.com/sourcegraph/conc/pool"
```

Comment:
This library has a useful result pool, kind of like of [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled). If it's a no go then I'd be happy to put something together with errgroup or similar.